### DISCUSSION TITLE
If an xpath corresponds to multiple elements on a page, how do you know the number of those elements

### DISCUSSION THREAD

**Discussion by chenqinggang001** [seleniumbase/SeleniumBase#1401] [Q&A] [answered]
I want to know the number of elements and click on them

**Comment by mdmintz** [comment, marked as answer]
Example with ``button``:

Find the number of ``button`` elements on a page:

``len(self.find_visible_elements("//button"))``

To click all the buttons:

``self.click_visible_elements("//button")``

**Comment by chenqinggang001** [comment]
Thanks

**Comment by chenqinggang001** [comment]
I need to click the buttons in order, The way I do it now is I get the length and I concatenate the elements in order, Is there a more elegant way to do it
@mdmintz  
The way I use it now:
```
for i in range(1, len(self.find_visible_elements("//button"))):
    self.click(f'("//button")[{i}]', timeout=10, delay=1)
    self.sleep(10)
```
Because I do something after every click, this method is not for me
`self.click_visible_elements("//button")`

**Reply by mdmintz** [reply]
If you have to perform an action between the clicks, then your way should work. I recommend changing the ``sleep()`` to waiting for an actual element, as that would save time from unnecessary sleeping/waiting.

**Reply by chenqinggang001** [reply]
ok，Thank you for reply
